engine/camunda/src/main/java/org/camunda/bpm/delegate/InterfaceWfcHandler.java

```diff
@@ -0,0 +1,174 @@
+    while (matcher.find()) {
+        String variableName = matcher.group(1);
+        logger.info("The required variable is: " + variableName);
+        String replacement;
+        if (variableName.equals("NOW") || variableName.startsWith("MOMENT")) {
+            logger.info("The moment in variable is: " + variableName);
+            replacement = "$(" + variableName + ")";
+        } else {
+            replacement = (String) delegateTask.getVariable(variableName);
+        }
+        matcher.appendReplacement(sb, Matcher.quoteReplacement(replacement));
+    }
+    matcher.appendTail(sb);
+    return sb.toString();
+}
+
+/* 
+ *  This method is used to handle the boundary message event interupting message boundary event attached to a user task
+ * @param messageBoundaryEvent 
+ * @param delegateTask
+*/
+public void handleBoundaryEvent(BoundaryEvent messageBoundaryEvent, DelegateTask delegateTask,
+        Properties properties) {
+    MessageEventDefinition messageEventDefinition = (MessageEventDefinition) messageBoundaryEvent
+            .getEventDefinitions().iterator().next();
+    Message message = messageEventDefinition.getMessage();
+    String messageName = message != null ? message.getName() : "No message";
+
+    Collection<Documentation> documentations = messageBoundaryEvent.getDocumentations();
+    String documentationText = !documentations.isEmpty() ? documentations.iterator().next().getTextContent()
+            : "No documentation";
+
+    logger.info("Handling boundary event ID from separate component: " + messageBoundaryEvent.getId()
+            + " with message: " + messageName);
+    String variableName = messageBoundaryEvent.getName();
+    String query = processQuery(documentationText, delegateTask);
+    requestData(query, delegateTask.getProcessInstanceId(), messageName, variableName, delegateTask.getId(),
+            properties);
+}
+/*
+ * This method is used to handle the data object reference in this implemetation associated with receieve task
+ * @param properties, delegateExecution
+ */
+public void dataObjectReferenceImpl(Properties properties, DelegateExecution delegateExecution) {
+    String message = "";
+    ReceiveTask receiveTask = null;
+    receiveTask = delegateExecution.getProcessEngine().getRepositoryService()
+                .getBpmnModelInstance(delegateExecution.getProcessDefinitionId())
+                .getModelElementById(delegateExecution.getCurrentActivityId());
+    message = receiveTask.getMessage().getName();
+    String variableName = "";
+    String query = "";
+        BpmnModelInstance bpmModel = delegateExecution.getProcessEngine().getRepositoryService()
+                .getBpmnModelInstance(delegateExecution.getProcessDefinitionId());
+        receiveTask = bpmModel.getModelElementById(delegateExecution.getCurrentActivityId());
+        // Get documentation from dataReference, NOTE: Assumed only 1 dataRef with a single documentation is found.
```

Comment:
Why the assumption that only 1 documentation (dataRef) is found ?

Reply:
This is not my assumption; it was there originally.